src/doom/p_inter.c

```diff
@@ -802,27 +799,12 @@ P_KillMobj
     // Drop stuff.
     // This determines the kind of object spawned
     // during the death frame of a thing.
-    switch (target->type)
+    if (target->info->droppeditem != MT_NULL)
```

Comment:
I'd suggest something like this to preserve the logic of the original code:
```
if (target->info->droppeditem != MT_NULL)
{
    item = target->info->droppeditem;
}
else
    return;

```

and keep the code as it was from then on.